quic/s2n-quic-core/src/crypto/tls/null.rs

```diff
@@ -0,0 +1,407 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Minimal endpoint that doesn't perform any handshakes or encryption
+//!
+//! NOTE: this should only be used for internal testing
+//!
+//! The main purpose of this endpoint type is to identify bottlenecks in the code _outside_ of
+//! crypto, since that's a large portion of the cycles we use. This has similar goals to the
+//! proposal in https://datatracker.ietf.org/doc/html/draft-banks-quic-disable-encryption-00, but
+//! instead reduces the handshake to simply exchanging transport parameters.
+
+use crate::{
+    application::{server_name::LOCALHOST, ServerName},
+    crypto::{self, tls},
+    transport,
+};
+use bytes::Bytes;
+use core::{mem::size_of, task::Poll};
+
+#[derive(Default, Debug)]
+pub struct Endpoint(());
+
+impl crypto::tls::Endpoint for Endpoint {
+    type Session = Session;
+
+    #[inline]
+    fn new_server_session<Params: s2n_codec::EncoderValue>(
+        &mut self,
+        transport_parameters: &Params,
+    ) -> Self::Session {
```

Comment:
my worry is if we're testing the connection creation latency outside of crypto it would also affect the measurement since we're printing to the console

Reply:
Actually thinking about it a bit, I'll put it in the `Endpoint::default()` method, since that's only created at start up.